allensdk/brain_observatory/behavior/data_objects/projections.py

```diff
@@ -0,0 +1,151 @@
+from matplotlib import image as mpimg
+from pynwb import NWBFile, ProcessingModule
+from pynwb.base import Images
+from pynwb.image import GrayscaleImage
+
+from allensdk.brain_observatory.behavior.data_objects import DataObject
+from allensdk.brain_observatory.behavior.data_objects.base \
+    .readable_interfaces import \
+    JsonReadableInterface, NwbReadableInterface, \
+    InternalReadableInterface
+from allensdk.brain_observatory.behavior.data_objects.base \
+    .writable_interfaces import \
+    NwbWritableInterface
+from allensdk.brain_observatory.behavior.image_api import ImageApi, Image
+from allensdk.brain_observatory.nwb.nwb_utils import get_image
+from allensdk.internal.api import PostgresQueryMixin
+from allensdk.internal.core.lims_utilities import safe_system_path
+
+
+class Projections(DataObject, InternalReadableInterface, JsonReadableInterface,
+                  NwbReadableInterface, NwbWritableInterface):
+    def __init__(self, max_projection: Image, avg_projection: Image):
+        super().__init__(name='projections', value=self)
+        self._max_projection = max_projection
+        self._avg_projection = avg_projection
+
+    @property
+    def max_projection(self) -> Image:
+        return self._max_projection
+
+    @property
+    def avg_projection(self) -> Image:
+        return self._avg_projection
+
+    @classmethod
+    def from_internal(cls, ophys_experiment_id: int,
+                      lims_db: PostgresQueryMixin) -> "Projections":
+        query = """
+                SELECT
+                    wkf.storage_directory || wkf.filename AS filepath,
+                    wkft.name as wkfn
+                FROM ophys_experiments oe
+                JOIN ophys_cell_segmentation_runs ocsr
+                ON ocsr.ophys_experiment_id = oe.id
+                JOIN well_known_files wkf ON wkf.attachable_id = ocsr.id
+                JOIN well_known_file_types wkft
+                ON wkft.id = wkf.well_known_file_type_id
+                WHERE ocsr.current = 't'
+                AND wkf.attachable_type = 'OphysCellSegmentationRun'
+                AND wkft.name IN ('OphysMaxIntImage',
+                    'OphysAverageIntensityProjectionImage')
+                AND oe.id = {};
+                """.format(ophys_experiment_id)
+        res = lims_db.select(query=query)
+        res['filepath'] = res['filepath'].apply(safe_system_path)
+
+        def _get_pixel_size():
```

Comment:
why is this one in an internal function, but the filepath query is not?

Reply:
I will update to place the filepath query in an inner function as well